### DISCUSSION TITLE
I want the lines to gradient on both sides. How should I write the code?

### DISCUSSION THREAD

**Discussion by junmaqiang** [fyne-io/fyne#4230] [Q&A]
# I want a gradient with black in the middle and white on both sides

```
// bg := canvas.NewRasterWithPixels(rgbGradient)
// bg := canvas.NewHorizontalGradient(color.Black, color.White)
// bg := canvas.NewVerticalGradient(color.White, color.White)
bg := canvas.NewRadialGradient(
color.NRGBA{R: 26, G: 188, B: 156, A: 255}, color.NRGBA{0xff, 0xff, 0xff, 0xff},
)
// bg := canvas.NewRectangle(color.Black)


// func rgbGradient(x, y, w, h int) color.Color {
// 	g := int(float32(x) / float32(w) * float32(255))
// 	b := int(float32(y) / float32(h) * float32(255))
// 	return color.NRGBA{uint8(255 - b), uint8(g), uint8(b), 0xff}
// }

```

**Comment by junmaqiang** [comment]
#  It's barely possible, but there's a gap in the middle
```
// NewLineGradientLeftRight 
func NewLineGradientLeftRight(s, e color.Color) *fyne.Container {
	
	left := canvas.NewHorizontalGradient(e, s)
	left.SetMinSize(fyne.NewSize(107, 5))
	//
	right := canvas.NewHorizontalGradient(s, e)
	right.SetMinSize(fyne.NewSize(107, 5))
	// 
	con := container.NewHBox(left, right)
	// 
	return container.NewBorder(Gao(10), Gao(10), nil, nil, con)
}
```
<img width="217" alt="截屏2023-09-10 22 37 21" src="https://github.com/fyne-io/fyne/assets/1819457/a2539f8d-384d-4062-a12a-8398d6de8887">

**Reply by andydotxyz** [reply]
There is a gap in the middle because you're using a container - they put spacing between widgets

**Reply by Jacalz** [reply]
Is it not the use of layout that causes the gap, and not the use of a container? I'd say that the solution is to create a layout that doesn't add padding in between and just place the gradients next to each other.

**Reply by andydotxyz** [reply]
Good point, I guess I should have said "a standard layout" and not "a container".

A custom layout would indeed solve this and the maths is pretty easy for a 50/50 split :).

**Reply by junmaqiang** [reply]
# What writing method should I use instead?
`con := container.NewGridWithColumns(2, left, right)`

**Reply by Jacalz** [reply]
See https://developer.fyne.io/extend/custom-layout
